### PR TITLE
Added support for span linking

### PR DESCRIPTION
**Pull request by Shachi16**

### CODE REVIEW

File: packages/jaeger-ui-components/src/url/ReferenceLink.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export default function ReferenceLink(props: ReferenceLinkProps) {
         if (!createLinkToExternalSpan) {
           throw new Error("ExternalLinkContext does not have a value, you probably forgot to setup it's provider");
         }
+        console.log(createLinkToExternalSpan(reference.traceID, reference.spanID))
```
Comment (erikShopify) [inline packages/jaeger-ui-components/src/url/ReferenceLink.tsx]: rm console.log :)

**Comment by erikShopify** [comment]
Also. add `var fields = []string{
	"traceID",
	"spanID",
	"parentSpanID",
	"operationName",
	"serviceName",
	"serviceTags",
	"startTime",
	"duration",
	"logs",
	"references",
	"tags",
}`

to file trace_transform_test.go at the bottom of the file to get all tests passing 💯

File: packages/jaeger-ui-components/src/url/ReferenceLink.tsx (hunk 1 above)
Comment (Shachi16) [reply packages/jaeger-ui-components/src/url/ReferenceLink.tsx]: Good catch!

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.tsx [hunk 2]
```diff
@@ -344,7 +350,7 @@ export class UnthemedSpanBarRow extends React.PureComponent<SpanBarRowProps> {
     rpc: null,
   };
 
-  _detailToggle = () => {
+  _detailToggle = async () => {
```
Comment (erin-noe-payne) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.tsx]: is this async?

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx [hunk 3]
```diff
@@ -14,18 +14,51 @@
...
+import { uAlignIcon, ubMb1 } from '../../uberUtilityStyles';
 
-const getStyles = createStyle(() => {
+const getStyles = createStyle((theme: Theme) => {
   return {
+    AccordianReferenceItem: css`
+      border-bottom: 1px solid ${autoColor(theme, '#d8d8d8')};
+    `,
+    AccordianKeyValues: css`
+      margin-left: 20px;
```
Comment (erin-noe-payne) [inline packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx]: based on your screenshots in slack, idk if this margin left is needed. The expand arrow makes it feel appropriately "indented" already

**Review by erin-noe-payne — APPROVED** [review]
Tiny UX feedback, looks great. We should pair to squash commits before we merge into `shopify-main`

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx (hunk 3 above)
Comment (Shachi16) [reply packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx]: Screen Shot 2022-01-25 at 2 32 32 PM

This is what it looks like without the margin, personally I think it looks better with some margin, maybe I should adjust to 10px if you think 20 is too much?

Comment (Shachi16) [reply packages/jaeger-ui-components/src/TraceTimelineViewer/SpanDetail/AccordianReferences.tsx]: Screen Shot 2022-01-25 at 2 37 26 PM
10px

File: packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.tsx (hunk 2 above)
Comment (Shachi16) [reply packages/jaeger-ui-components/src/TraceTimelineViewer/SpanBarRow.tsx]: No that was added when I was messing around when trying to get the references to open, this shouldn't have been committed could catch

**Pull request merged**